src/eval/StateService.ml

```diff
@@ -40,6 +40,9 @@ type ss_field = {
   fval : SSLiteral.t option; (* We may or may not have the value in memory. *)
 }
 
+(* The blockchain info is a map from (query_name, query_args) to some info. *)
+type bcinfo_state = (string, (string, string) Caml.Hashtbl.t) Caml.Hashtbl.t
```

Comment:
As mentioned above I don't understand why you need to map `BLOCKNUMBER` to a hashtable.

Reply:
> 

So that we can have other blockchain queries with arguments and the arguments form the second key.

Reply:
OK. I guess this works, but the timestamp feature will probably guide us.